include/ginkgo/core/base/mpi.hpp

```diff
@@ -1619,16 +1619,19 @@ class window {
      * @param lock_t  the type of the lock: shared or exclusive
      * @param assert  the optimization level. 0 is always valid.
      */
-    void lock(int rank, lock_type lock_t = lock_type::shared, int assert = 0)
+    void lock(int rank, lock_type lock_t = lock_type::shared,
+              int assert = 0) const
     {
-        if (lock_t == lock_type::shared) {
-            GKO_ASSERT_NO_MPI_ERRORS(
-                MPI_Win_lock(MPI_LOCK_SHARED, rank, assert, this->window_));
-        } else if (lock_t == lock_type::exclusive) {
-            GKO_ASSERT_NO_MPI_ERRORS(
-                MPI_Win_lock(MPI_LOCK_EXCLUSIVE, rank, assert, this->window_));
-        } else {
-            GKO_NOT_IMPLEMENTED;
+        if (&this->window_) {
```

Comment:
What is this check actually there for? `window_` is always initialized and has always a valid value. I think you want to test against `MPI_WIN_NULL`?

Reply:
Okay, I just verified that both `MPI_WIN_NULL` and `nullptr` when passed through the mpi error checking throws an MPI error, which is good enough. So I have removed these checks now.